modules/vaos/app/services/vaos/user_service.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+module VAOS
+  class UserService < Common::Client::Base
+    configuration VAOS::Configuration
+
+    def session(user)
+      cached = SessionStore.find(user.uuid)
+      return cached.token if cached
+
+      create_session(user)
+    end
+
+    private
+
+    def create_session(user)
+      url = '/users/v2/session?processRules=true'
+      token = VAOS::JWT.new(user).token
+      response = perform(:post, url, token, 'Content-Type' => 'text/plain', 'Referer' => 'https://api.va.gov')
+      if response&.body
+        SessionStore.new(user_uuid: user.uuid, token: response.body).save
+        response.body
+      end
+    end
```

Comment:
this doesn't appear to handle exceptions; if the the service is down or unavailable will this be invoked indefinitely?

Reply:
Ya the Faraday error gets caught by the common client and that error bubbles up to the controller that was trying to use the token. Then the application controller error handling would return a json error response. Breakers also tracks metrics and, as you mentioned, would kick in if enough calls failed. In a later PR we can add VAOS specific errors to exceptions.en.yml and could add additional metrics.